groupnews-service/src/main/java/com/epam/internship/impl/UserServiceImpl.java

```diff
@@ -1,32 +1,65 @@
 package com.epam.internship.impl;
 
-import java.util.ArrayList;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.core.convert.TypeDescriptor;
 import org.springframework.stereotype.Service;
 
 import com.epam.internship.UserService;
 import com.epam.internship.dto.User;
+import com.epam.internship.entity.UserEntity;
+import com.epam.internship.repo.UserRepository;
 
 @Service
 public class UserServiceImpl implements UserService {
+
+	@Autowired
+	private UserRepository userRepository;
+
+	@Autowired
+	private ConversionService conversionService;
+
+	private final String EMAIL_PATTERN = "^[_A-Za-z0-9-\\+]+(\\.[_A-Za-z0-9-]+)*@"
+			+ "[A-Za-z0-9-]+(\\.[A-Za-z0-9]+)*(\\.[A-Za-z]{2,})$";
 	
-	private List<User> users = new ArrayList<>();
+	private Pattern pattern = Pattern.compile(EMAIL_PATTERN);
 	
-	public UserServiceImpl() {
-		users.add(User.builder().name("Mr. Brown").email("brown@email.com").build());
-	}
-
 	@Override
 	public User createUser(String name, String email) {
 		User newUser = User.builder().name(name).email(email).build();
-		users.add(newUser);
+		userRepository.save(conversionService.convert(newUser, UserEntity.class));
 		return newUser;
 	}
 
+	@SuppressWarnings("unchecked")
 	@Override
 	public List<User> getAllUsers() {
-		return users;
+		return (List<User>) conversionService.convert(userRepository.findAll(),
+				TypeDescriptor.collection(List.class, TypeDescriptor.valueOf(UserEntity.class)),
+				TypeDescriptor.collection(List.class, TypeDescriptor.valueOf(User.class)));
+
+	}
+
+	@Override
+	public boolean emailAlreadyExists(String email) {
+		for(UserEntity userEntity : userRepository.findAll()){
+			if(userEntity.getEmail().equals(email))
```

Comment:
It is recommended to use curly braces ({}) even if there is only a single statement in the body of the if / for. This improves readability a lot.